File: app/src/main/java/com/recursia/popularmovies/utils/intro/PrefUtilsImpl.java
package com.recursia.popularmovies.utils.intro;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class PrefUtilsImpl implements PrefUtils {
    private static final String TAG = "intro";
    private final Context context;

    public PrefUtilsImpl(Context context) {
        this.context = context;
    }

    @Override
    public boolean getValue() {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        return preferences.getBoolean(TAG, true);
    }

    @Override
    public void putValue(boolean value) {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        preferences.edit().putBoolean(TAG, value).apply();
    }

}

File: app/src/main/java/com/recursia/popularmovies/utils/intro/PrefUtilsImpl.kt
package com.recursia.popularmovies.utils.intro

import android.content.Context
import android.preference.PreferenceManager

class PrefUtilsImpl(private val context: Context) : PrefUtils {

    override val value: Boolean
        get() {
            val preferences = PreferenceManager.getDefaultSharedPreferences(context)
            return preferences.getBoolean(TAG, true)
        }

    override fun putValue(value: Boolean) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
        preferences.edit().putBoolean(TAG, value).apply()
    }

    companion object {
        private const val TAG = "intro"
    }

}
